app/models/variant_detail.rb

```diff
@@ -1,14 +1,16 @@
-class VariantDetail
-  attr_reader :split, :variant, :weight, :assignment_count
-
-  def initialize(split, variant)
-    @split = split
-    @variant = variant
-    @weight = split.variant_weight(variant)
-    @assignment_count = split.assignment_count_for_variant(variant)
+class VariantDetail < ActiveRecord::Base
+  belongs_to :split
+
+  validate :variant_must_exist, on: :create
```

Comment:
Or, if we're really trying to optimize when this runs, it would apply on change of variant. But I'd be fine with just saying it applies always.